pages/nfts/new.tsx

```diff
@@ -224,61 +224,59 @@ export default function BulkUploadWizard() {
     }
   };
 
-  const transformFormVals = async (values: NFTFormValue[], filePins: UploadedFilePin[]) => {
-    const resp = await Promise.all(
-      values.map(async (v, i: number) => {
-        const filePin = filePins[i];
-
-        if (!filePin) {
-          throw new Error('No file pin for index ' + i);
-        }
+  const transformFormVal = async (
+    value: NFTFormValue,
+    index: number,
+    filePins: UploadedFilePin[]
+  ) => {
+    const filePin = filePins[index];
+
+    if (!filePin) {
+      throw new Error('No file pin for index ' + index);
+    }
 
-        const category: NFTCategory = detectCategoryByFileExt(filePin.name);
-        const isMultiMedia = category !== 'image';
+    const category: NFTCategory = detectCategoryByFileExt(filePin.name);
+    const isMultiMedia = category !== 'image';
 
-        let coverImageFile: UploadedFilePin | undefined;
-        if (v.coverImageFile) {
-          coverImageFile = await uploadCoverImage(v.coverImageFile);
-        }
+    let coverImageFile: UploadedFilePin | undefined;
+    if (value.coverImageFile) {
+      coverImageFile = await uploadCoverImage(value.coverImageFile);
```

Comment:
@kespinola When you say this should be moved out of this transform, do you have an idea where?  As in we should store the filePins for coverImages earlier then retrieve them based on index?

Reply:
So my thought was when they select a cover image on the info page they may hit the back button and make a change to that image, so we only upload the coverImage to IPFS when we are about to send off the metadata and avoid putting things on IPFS unless absolutely needed.